### ISSUE TITLE
Once I close Slippi, I can no longer open it again with the controller button input unless I run the .exe again manually

### ISSUE THREAD

**Issue by davidvkimball** [RucksP/Instant_Melee#2]
This may not be an issue, so apologies if it isn't, but once I close out of Slippi, I can't seem to launch it again by pressing start on my controller. The only way I can is if I run instant_melee.exe again, is this intended? If so, is there a way for it to automatically re-launch upon Slippi close?

**Comment by RucksP** [comment]
Can you send a copy of what logs.txt says on your end, it works fine for me. The program should never close, it just stalls while the dolphin process it creates is open.

The only thing i could think of that causes this is if you open an instance of dolphin by m/kb instead of your gc controller. That'll cause the program to close due to not having access to the adapter.

**Comment by davidvkimball** [comment]
> 
> 
> Can you send a copy of what logs.txt says on your end, it works fine for me. The program should never close, it just stalls while the dolphin process it creates is open.
> 
> The only thing i could think of that causes this is if you open an instance of dolphin by m/kb instead of your gc controller. That'll cause the program to close due to not having access to the adapter.

So I realized, it's probably just because I've opened another instance of Dolphin also. I use Slippi but also use Dolphin, so maybe that's confusing it? Even when I close the other instance, it stops working. And sure, here it is: [logs.txt](https://github.com/RucksP/Instant_Melee/files/5667823/logs.txt)

**Comment by RucksP** [comment]
The TL;DR is if you close instant_melee via task manager, then close all programs which try to access your GC adapter (slippi or mainline dolphin), then open instant_melee, and then never open another program which tries to take control of your adapter by m/kb it should work fine.

I think the instant_melee is working as intended. it's just opening 2 dolphins kinda messes with it. **Only 1 program can have access to the adapter at a time**. The way it should normally work is:
1. Connect to the adapter
2. Wait for start to be pressed
3. Disconnect from the adapter (so dolphin can take control)
4. Open Dolphin
5. Check if the dolphin process opened by instant_melee is still open
6. If not, connect to the adapter again and go to step 2.

Opening dolphin through another method before step 1 or during step 2 will cause the program to close due to not having access to the adapter when it should.

It does not have access in case 1 because the dolphin process already took control of the adapter.

In the 2nd case it has control until you press start, in which case a new dolphin will open, and the first dolphin opened will then have access to the adapter. After the 2nd dolphin is closed instant_melee will then try to access the adapter again, but will be unable to because the dolphin opened by m/kb will have control.

**Comment by davidvkimball** [comment]
Thank you for the thoughtful and thorough explanation! You didn't have to and I appreciate your insight. :)
